lib/restbase.js

```diff
@@ -39,6 +40,11 @@ function RESTBase(options, req) {
         this._recursionDepth = par._recursionDepth + 1;
         this._priv = par._priv;
         this.rb_config = this._priv.options.conf;
+        this._rootReq = par._rootReq || req;
```

Comment:
For performance, it is better to make sure that properties are assigned in the same order in both branches, so that the object shape ends up being the same.